scripts/irods/test/test_iadmin.py

```diff
@@ -1456,6 +1456,69 @@ def test_print_full_username__issue_3170(self):
         self.admin.assert_icommand('iadmin lu {0}'.format(username), 'STDOUT', 'user_name: {0}'.format(username))
         self.admin.assert_icommand('iadmin rmuser {0}'.format(username))
 
+    def test_empty_data_mode_does_not_cause_INVALID_LEXICAL_CAST_on_rebalance__issue_5227(self):
+        filename = os.path.join(self.admin.local_session_dir, 'issue_5227')
+        data_object = os.path.join(self.admin.session_collection, os.path.basename(filename))
+
+        try:
+            # Create resources.
+            self.admin.assert_icommand(['iadmin', 'mkresc', 'root_resc', 'passthru'], 'STDOUT', ['root_resc'])
+            self.admin.assert_icommand(['iadmin', 'mkresc', 'pt0', 'passthru'], 'STDOUT', ['pt0'])
+            self.admin.assert_icommand(['iadmin', 'mkresc', 'pt1', 'passthru'], 'STDOUT', ['pt1'])
+            self.admin.assert_icommand(['iadmin', 'mkresc', 'repl_resc', 'replication'], 'STDOUT', ['repl_resc'])
+            lib.create_ufs_resource('ufs0', self.admin)
+            lib.create_ufs_resource('ufs1', self.admin)
+
+            # Create resource hierarchy.
+            self.admin.assert_icommand(['iadmin', 'addchildtoresc', 'root_resc', 'repl_resc'])
+            self.admin.assert_icommand(['iadmin', 'addchildtoresc', 'repl_resc', 'pt0'])
+            self.admin.assert_icommand(['iadmin', 'addchildtoresc', 'repl_resc', 'pt1'])
+            self.admin.assert_icommand(['iadmin', 'addchildtoresc', 'pt0', 'ufs0'])
+            self.admin.assert_icommand(['iadmin', 'addchildtoresc', 'pt1', 'ufs1'])
+
+            # Create a new data object in the hierarchy.
+            lib.make_file(filename, 0)
+            self.admin.assert_icommand(['iput', '-R', 'root_resc', filename, data_object])
+            self.admin.assert_icommand(['ils', '-l', data_object], 'STDOUT', [' root_resc;repl_resc;pt0;ufs0 ', ' root_resc;repl_resc;pt1;ufs1 '])
+
+            # While the issue was encountered via an empty string, we don't have any tools that
+            # allow us to create the empty string scenario. Instead, we know the empty string scenario
+            # is not any different then the data mode containing a non-integer value. Therefore, using
+            # an invalid string for the data mode satisfies the test.
+            for invalid_data_mode in [' ', '-', 'X']:
+                # Set the source replica's data mode to an invalid value.
+                self.admin.assert_icommand(['iadmin', 'modrepl', 'logical_path', data_object, 'replica_number', '0', 'DATA_MODE', invalid_data_mode])
+
+                # Create work for the rebalance operation by setting the second replica's status to stale.
```

Comment:
Would it be worth adding a section to this test to exercise the "create missing replicas" codepath? I realize the change occurs in a common section, but it could be worthwhile for completeness.

Reply:
Sure. Adding that case will protect us from breaking things in the future.

Reply:
done.